.eslintrc.js

```diff
@@ -52,6 +52,13 @@ module.exports = {
       },
     ],
     '@typescript-eslint/consistent-type-imports': 'error',
+    'no-restricted-syntax': [
+      'error',
+      {
+        selector: 'AssignmentExpression [name=defaultProps]',
+        message: 'defaultProps are not allowed, use function defaults instead.',
+      },
+    ],
```

Comment:
Wondering if this needed now that we don't have any occurrences since it's not a common pattern?

Reply:
I accidentally used it in new code yesterday, so it's better to have it for a bit, especially when there's more class-to-function refactors to come where it could be forgotten to migrate